src/main/java/org/wso2/extension/siddhi/window/uniquetimebatch/UniqueTimeBatchWindowProcessor.java

```diff
@@ -0,0 +1,335 @@
+import org.wso2.siddhi.core.query.processor.stream.window.FindableProcessor;
+import org.wso2.siddhi.core.query.processor.stream.window.WindowProcessor;
+import org.wso2.siddhi.core.table.EventTable;
+import org.wso2.siddhi.core.util.Scheduler;
+import org.wso2.siddhi.core.util.collection.operator.Finder;
+import org.wso2.siddhi.core.util.collection.operator.MatchingMetaStateHolder;
+import org.wso2.siddhi.core.util.parser.OperatorParser;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.exception.ExecutionPlanValidationException;
+import org.wso2.siddhi.query.api.expression.Expression;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * UniqueTimeBatch window
+ *
+ * @since 1.0.0
+ */
+public class UniqueTimeBatchWindowProcessor extends WindowProcessor implements SchedulingProcessor,
+        FindableProcessor {
+
+    private long timeInMilliSeconds;
+    private long nextEmitTime = -1;
+    private ComplexEventChunk<StreamEvent> currentEventChunk = new ComplexEventChunk<>(false);
+    private ComplexEventChunk<StreamEvent> eventsToBeExpired = null;
+    private ConcurrentHashMap<String, StreamEvent> oldEventMap = new ConcurrentHashMap<>();
+    private StreamEvent resetEvent = null;
+    private Scheduler scheduler;
+    private ExecutionPlanContext executionPlanContext;
+    private boolean isStartTimeEnabled = false;
+    private long startTime = 0;
+    private VariableExpressionExecutor[] variableExpressionExecutors;
+
+    /**
+     * The setScheduler method of the TimeWindowProcessor, As scheduler is private variable,
+     * to access publicly we use this setter method.
+     *
+     * @param scheduler the value of scheduler.
+     */
+    @Override
+    public void setScheduler(Scheduler scheduler) {
+        this.scheduler = scheduler;
+    }
+
+    /**
+     * The getScheduler method of the TimeWindowProcessor, As scheduler is private variable,
+     * to access publicly we use this getter method.
+     */
+    @Override
+    public Scheduler getScheduler() {
+        return scheduler;
+    }
+
+    /**
+     * The init method of the WindowProcessor, this method will be called before other methods.
+     *
+     * @param attributeExpressionExecutors the executors of each function parameters
+     * @param executionPlanContext         the context of the execution plan
+     */
+    @Override
+    protected void init(ExpressionExecutor[] attributeExpressionExecutors,
+                        ExecutionPlanContext executionPlanContext) {
+        this.executionPlanContext = executionPlanContext;
+        variableExpressionExecutors = new VariableExpressionExecutor[attributeExpressionExecutors.length - 1];
+        this.eventsToBeExpired = new ComplexEventChunk<>(false);
+        if (attributeExpressionExecutors.length == 2) {
+            variableExpressionExecutors[0] = (VariableExpressionExecutor) attributeExpressionExecutors[0];
```

Comment:
I'm wondering why we're not assigning this to a VariableExpressionExecutor variable. Any reason to use an array?